compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt

```diff
@@ -239,6 +240,14 @@ open class ClassCodegen protected constructor(
                     }
                 }
             }
+            is MetadataSource.Function -> {
+                val functionProto = serializer!!.functionProto(metadata.descriptor)?.build()
```

Comment:
I suppose you can call `createFreeFakeLambdaDescriptor` on this descriptor (as in `ClosureCodegen.generateKotlinMetadataAnnotation`) to also support `functionNtoStringGeneric.kt`.